src/components/Home/Highlights/HighlightsManager.tsx

```diff
@@ -0,0 +1,131 @@
+     */
+    id: number
+    /**
+     * enum field to identify the type of the widget
+     */
+    type: WidgetType
+    /**
+     * string to identify the date of the event(or deadline) contained in the widget
+     */
+    date: string
+    /**
+     * string to identify the time of the event(or deadline) contained in the widget
+     */
+    time: string
+    /**
+     * string to identify the title of the event contained in the widget
+     */
+    title: string
+    /**
+     * string to identify eventually the room when the event takes place
+     */
+    room?: string
+}
+
+/**
+ * Component that decides the content of the carousel.
+ * In the future the logic bheind the highlights will be implemented here and it will pass
+ * the correct data to the PolimiCarousel
+ */
+export const HighlightsManager: FC = () => {
+    const [isLessonFound, setIsLessonFound] = useState(false)
+    const [data, setData] = useState<CarouselItem[]>([
+        {
+            id: 0,
+            type: WidgetType.ISEE,
+            date: "mer 22/10/2022",
+            time: "8:00",
+            title: "Scadenza ISEE 2023",
+        },
+        {
+            id: 1,
+            type: WidgetType.ISEE,
+            date: "mer 22/10/2022",
+            time: "8:00",
+            title: "Scadenza ISEE 2023",
+        },
+        {
+            id: 2,
+            type: WidgetType.ISEE,
+            date: "mer 22/10/2022",
+            time: "8:00",
+            title: "Scadenza ISEE 2023",
+        },
+    ])
+
+    const findNextLecture = (timetable: Lecture[]) => {
+        /**
+         * sorting dates in ascending order, priority is the 'date_start' field.If there are two lecture with the same
+         * 'date_start' we check the 'date_end' field!
+         * @example
+         * lecture 1{date_start: "2022-12-04T08:15:00",date_end: "2022-12-04T13:15:00"}
+         * lecture 2{date_start: "2022-12-04T08:15:00",date_end: "2022-12-04T14:00:00"}
+         * So in this case lecture 1 has more priority than lecture 2
+         */
+        const sorted = timetable.sort((a, b) => {
+            let dateA = new Date(a.date_start).getTime()
+            let dateB = new Date(b.date_start).getTime()
+            if (dateA !== dateB) {
+                return dateA - dateB
+            } else {
+                dateA = new Date(a.date_end).getTime()
+                dateB = new Date(b.date_end).getTime()
+                return dateA - dateB
+            }
+        })
+        const currentDate = new Date().toISOString().slice(0, 10)
+        const currentTime = new Date().toLocaleTimeString() //in order to get the current local time
+        const now = currentDate + "T" + currentTime
+
+        for (let i = 0; i < sorted.length; i++) {
+            if (
+                new Date(sorted[i].date_start) > new Date(now) ||
+                (new Date(now) > new Date(sorted[i].date_start) &&
+                    new Date(now) < new Date(sorted[i].date_end))
```

Comment:
Correggetemi se sbaglio, stavo guardando il codice e ho pensato: al posto di fare cosi non si può direttamente fare `now = new Date()` ?

```suggestion
        const now = new Date()

        for (let i = 0; i < sorted.length; i++) {
            if (
                new Date(sorted[i].date_start) > now ||
                (now > new Date(sorted[i].date_start) &&
                    now < new Date(sorted[i].date_end))
```

Reply:
Ciao Diego, all’inizio lo avevo fatto così come dici  tu ma se non erro l’orario che ottenevo era quello del regno unito(un’ora in meno rispetto al nostro)!In questo modo avevo aggirato il problema !

Reply:
ma in teoria dovrebbe essere giusto avere l'orario segnato un'ora indietro no? gli orari dagli endpoint del poli vengono restituiti nel formato locale tipo `2022-12-08T08:15:00` che sono le 8:15 ora italiana
```ts
new Date("2022-12-08T08:15:00").toISOString()
```
ottieni `2022-12-08T07:15:00.000Z` dove la Z significa orario UTC+0.
così come se fai
```ts
new Date().toISOString()
```
ottieni `2022-12-08T21:48:14.278Z` ipotizzando che siano le 22:48

morale, l'offset dovrebbe essere sempre di un'ora che tu stia Date.now() o che tu stia usando la stringa del poli no? (e 2 durante l'estate)

Reply:
Okay si ha senso, mi ero un attimo confuso con le convenzioni :) grazie tommaso e diego